### PR TITLE
allow individual maxLimit config in rest service monitor

### LINKED ISSUE #1033: Allow for configuration of specific checks on the monitor
**Problem**
The current monitor checks assume a high traffic environment with many accounts (1000+). These conditions are not always present in networks other than the Hedera Mainnet or Public testnet.

**Solution**
Make specific checks and thresholds configurable. The specific checks which I've found problematic are the following:

- transactions.length of 174 is less than limit 1000
- transactions.length of 174 is less than limit 1000
- balances.length of 106 is less than limit 1000
- accounts.length of 106 is less than limit 1000

Occasionally the data freshness check of 50 seconds can also cause issues, but the check is very high value and there is a workaround (automated periodic transfers) that works well to address that, so I'm less concerned with that check.

**Alternatives**
Rather than going through the trouble of making each individual check and/or threshold configurable, we could have a single dev or low utilization mode which just omits those checks entirely.

### PR DESCRIPTION
**Pull request by xin-hedera**
**Detailed description**:

Supports per-resource `maxLimit` config in rest service monitor, so ops can configure a smaller threshold for low traffic environments.

- Adds separate configuration option `maxLimit` for `account`, `balance`, and `transaction`. The value will overrides the global `maxLimit` if it's smaller
- Adds missing consensus timestamp ascending order check for transactions query
- Removes dependency on hedera-mirror-rest/config and moves all configuration to `serverlist.json`
- Aggregates http response code from all servers, returns 409 if there is at least one failed test 

**Which issue(s) this PR fixes**:
Fixes #1033  #462 

**Special notes for your reviewer**:

Tested in perfnet.

```shell
$ curl http://localhost:9001/api/v1/status/performance-6557 | jq .
```

```json
{
  "ip": "127.0.0.1",
  "name": "performance-6557",
  "port": "6557",
  "results": {
    "testResults": [
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/balances?limit=100",
        "message": "Successfully called balances and performed account check",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/balances?limit=1",
        "message": "Successfully retrieved balance from with 1800 seconds ago",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/accounts?timestamp=gt%3A1600439399.310535&timestamp=lt%3A1600439401.310535&limit=1",
        "message": "Successfully called accounts with time and limit params",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/balances?timestamp=gt%3A1600439399.310535&timestamp=lt%3A1600439401.310535&limit=1",
        "message": "Successfully called balances with time and limit params",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions?limit=1",
        "message": "Successfully retrieved transactions from with 50 seconds ago",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/balances?account.id=0.0.1009",
        "message": "Successfully called balances and performed account check",
        "failureMessages": []
      },
      {
        "at": 1600439717483,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions?timestamp=gt%3A1600439702.316232&timestamp=lt%3A1600439704.316232&limit=1",
        "message": "Successfully called transactions with time and limit params",
        "failureMessages": []
      },
      {
        "at": 1600439717483,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions/0.0.2-1600439692-302595000",
        "message": "Successfully retrieved single transactions by id",
        "failureMessages": []
      },
      {
        "at": 1600439717483,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions?limit=10",
        "message": "Successfully called transactions with limit params only",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/accounts?account.id=1001&type=credit&limit=1",
        "message": "Successfully called accounts and performed account check",
        "failureMessages": []
      },
      {
        "at": 1600439717483,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions?order=asc&limit=102",
        "message": "Successfully called transactions with order params only",
        "failureMessages": []
      },
      {
        "at": 1600439717483,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/transactions?account.id=3&type=credit&limit=1",
        "message": "Successfully called transactions and performed account check",
        "failureMessages": []
      },
      {
        "at": 1600439717484,
        "result": "passed",
        "url": "http://127.0.0.1:6557/api/v1/accounts/0.0.1001",
        "message": "Successfully called accounts for single account",
        "failureMessages": []
      }
    ],
    "numPassedTests": 13,
    "numFailedTests": 0,
    "success": true,
    "message": "",
    "startTime": 1600439717483,
    "endTime": 1600439717888
  },
  "httpCode": 200
}
```
**Checklist**
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]

File: hedera-mirror-rest/monitoring/monitor_apis/transaction_tests.js [hunk 1]
```diff
@@ -169,32 +170,40 @@ const getTransactionsWithAccountCheck = async (server, classResults) => {
...
-      check = false;
+  let previousConsensusTimestamp = '0';
+  for (const txn of transactions) {
+    if (txn.consensus_timestamp <= previousConsensusTimestamp) {
+      currentTestResult.failureMessages.push('consensus timestamps are not in ascending order');
+      acctestutils.addTestResult(classResults, currentTestResult, false);
+      return;
     }
-    prevSeconds = acctestutils.secNsToSeconds(txn.seconds);
+    previousConsensusTimestamp = txn.consensus_timestamp;
```
Comment (xin-hedera) [inline hedera-mirror-rest/monitoring/monitor_apis/transaction_tests.js:206]: changed from the last revision, directly compare the timestamp string

File: hedera-mirror-rest/monitoring/monitor_apis/monitortest_utils.js [hunk 2]
```diff
@@ -17,10 +17,11 @@
  * limitations under the License.
  * ‍
  */
-const math = require('mathjs');
-const config = require('../../config.js');
-const fetch = require('node-fetch');
+
 const AbortController = require('abort-controller');
+const config = require('../../config');
```
Comment (steven-sheehy) [inline hedera-mirror-rest/monitoring/monitor_apis/monitortest_utils.js:22]: It's quite strange that we're using the config reader from `hedera-mirror-rest` and that the monitor_api has two different config formats (e.g. application.yaml and serverlist.json). We should remove this dependency and just use serverlist.json.

File: hedera-mirror-rest/monitoring/monitor_apis/account_tests.js [hunk 3]
```diff
@@ -66,67 +68,65 @@ const checkMandatoryParams = (entry) => {
  * @param {Object} classResults shared class results object capturing tests for given endpoint
  */
 const getAccountsWithAccountCheck = async (server, classResults) => {
-  var currentTestResult = acctestutils.getMonitorTestResult();
+  const currentTestResult = acctestutils.getMonitorTestResult();
+
+  const query = {};
+  const {maxLimit, isGlobal} = acctestutils.getMaxLimit(resource);
+  if (!isGlobal) {
```
Comment (steven-sheehy) [inline hedera-mirror-rest/monitoring/monitor_apis/account_tests.js:75]: This conditional logic seems unnecessary and breaks encapsulation of the maxLimit properties. We should always have a limit set and include that in the query, regardless if it's the global or not.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/monitoring/monitor_apis/monitortest_utils.js (hunk 2 above)
Comment (steven-sheehy) [reply hedera-mirror-rest/monitoring/monitor_apis/monitortest_utils.js:22]: Please also add defaults to serverlist.json

File: hedera-mirror-rest/monitoring/monitor_apis/account_tests.js (hunk 3 above)
Comment (xin-hedera) [reply hedera-mirror-rest/monitoring/monitor_apis/account_tests.js:75]: the flag is used to honor the default `maxLimit` if there is no resource specific `maxLimit`

the original test checks whether the rest service honors the `maxLimit` set in application.yml, hence `limit=?` query should not be added

**Review by steven-sheehy — APPROVED** [review]

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**